### DISCUSSION TITLE
Why does BackToTopButton exist only on Docs Page?

### DISCUSSION THREAD

**Discussion by parkgang** [facebook/docusaurus#7516] [Q&A] [answered]
If you have a lot of content on the page and you have scrolls, you can create a good UX by providing the Back to top button

However, we can see that docusaurus provides Back to top button only for Docs Page.

Is this an intended design? I think it's good to provide a back to top button on other pages.

**Comment by Josh-Cena** [comment, marked as answer]
It's partially intended, because for non-content pages, like a React page, it may not always make sense, so we didn't add it to `Layout`. If you want to add it to blog, I think it makes sense as well.

**Reply by parkgang** [reply]
What do you think about putting the Back to top button in the `Layout component` because even if you add it to the Layout, it is only displayed for pages with scrolls?

When I tested it, I think there is no problem with UX because the Back to top button is displayed only in one case when there is content and there is scroll.

**Reply by Josh-Cena** [reply]
I don't like it on the landing page, for example. It makes no sense there.

I'm okay to put it in layout, if we have a default-false prop that turns it on. WDYT?

**Reply by parkgang** [reply]
Yes, it's fine. I think it would be nice to be able to configure it in a place like `docusaurus.config.js`.

Or, it would be nice if you could select the pages to be excluded even if you select the option as a whole, or if it is open so that it can be applied only to the desired platform among `pages`, `docs`, `blog`.

**Reply by Josh-Cena** [reply]
That sounds like a use-case for swizzling, if you want that level of control.

**Reply by parkgang** [reply]
That's right, it was an opinion that came out thinking that it would be good if the extension was open.

If so, what do you think about determining whether to apply only to the entire page with `docusaurus.config.js` rather than props?

**Reply by Josh-Cena** [reply]
I don't think we want to populate `docusaurus.config.js` with these kinds of theme options. Swizzling should be your go-to, as I already said. I'm fine with moving BTTB to the layout and adding a prop, but nothing else.

**Reply by parkgang** [reply]
okay. Thank you for answer.

**Reply by Josh-Cena** [reply]
Feel free to send a PR to make that change, so we can have some code to discuss on!
